keycloak-extensions/src/test/java/org/alvearie/keycloak/KeycloakContainerTest.java

```diff
@@ -41,8 +48,19 @@
     private static final String USERNAME = "a";
     private static final String PASSWORD = "a";
     private static final String KC_CLIENT = "test";
+    private static final String AUTH_ENDPOINT = "/auth/realms/test/protocol/openid-connect/auth";
     private static final String TOKEN_ENDPOINT = "/auth/realms/test/protocol/openid-connect/token";
 
+    // per https://www.testcontainers.org/features/networking/#exposing-host-ports-to-the-container
+    // the host's port should be known prior to starting the container
+    private static MockWebServer mockFhirServer;
+    static {
+        mockFhirServer = new MockWebServer();
+        setupFHIRPortEnvVar(mockFhirServer.getPort());
+        Testcontainers.exposeHostPorts(mockFhirServer.getPort());
```

Comment:
this magic line make it so that the keycloak container can reach the host's mock server port using a magical hostname of `host.testcontainers.interal`